onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1058,13 +1058,29 @@ The bounding box coordinates corresponding to the selected indices can then be o
       .Input(0, "X", "An input tensor to hash.", "T1")
       .Output(0, "Y", "32-bit hash value.", "T2")
       .TypeConstraint("T1", {"tensor(uint32)", "tensor(int32)", "tensor(string)"}, "Constrain input type to unsigned or signed 32-bit integer tensor, or string tensor. It should be utf-8 encoded if using unicode.")
-      .TypeConstraint("T2", {"tensor(uint32)", "tensor(int32)"}, "Constrain output type to unsigned or signed 32-bit integer tensor.")
+      .TypeConstraint("T2", {"tensor(uint32)", "tensor(int32)"}, "Constrain output type to unsigned and signed 32-bit integer tensor.")
       .Attr(
           "seed",
           "Seed for the hashing algorithm, unsigned 32-bit integer, default to 0.",
           AttributeProto::INT,
           (int64_t)0LL)
+      .Attr(
+          "positive",
+          "If value is 1, output type is uint32_t, else int32_t. Default value is 1.",
+          AttributeProto::INT,
+          (int64_t)1LL)
       .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+        // type inference
+        auto positive_attr = ctx.getAttribute("positive");
+        bool is_positive =
+            positive_attr ? (static_cast<int>(positive_attr->i()) == 1 ? true : false) : true /* default value if attribute not present */;
+        auto selected_indices_type = ctx.getOutputType(0)->mutable_tensor_type();
```

Comment:
>selected_indices_type [](start = 13, length = 21)

the naming is copied from other op? name it "output_data_type"?

Reply:
sure, will do.